### PR TITLE
Add Issue templates and automation scripts

### LINKED ISSUE #773: Improve development experience/triage process
## What

Today in the semantic conventions repo we see the following "issues" happening often:

1. Issues are automatic assigned to maintainers, and given the limited bandwidth are not always looked after and become stale/no response
2. Issues/PRs are not clearly marked to which area/space it belongs too. 
3. Everything is pretty much manual, apart from the automatic assigned person on issue/PRs 
4. The process of requesting new conventions, making changes to existing is not clear and not clearly defined
5. The process to contribute to the conventions needs improvements

This then results in:

- The path to get an issue from open to done is not clearly defined, which probably frustrates users and don't leave a good impression to the outside world
- Maintainers can get overloaded since they are the ones having to take care of all of it
- Sometimes nobody feels "responsible" of taking care of things, since ownership is not clearly defined

## What can we do

It's hard to solve everything at once, but my proposal that I shared in #292 after looking at other repos like the collector core/contrib, spec and others outside our org is:

- Define areas which correspond to the same file names in our [attributes registry](https://github.com/open-telemetry/semantic-conventions/tree/main/model/registry)
  - Those areas then become labels in the repo (addresses #22)
  
- Define issue templates that requires the user to select one of the above `areas`:
  - Bugs:
  - Propose changes
  - Propose new conventions
  
- Create automations that assigns the correct label on issues after they are created, based on the `area` the user selected during issue creation

- Clearly document the process to contribute and what labels means and etc

I believe that by implementing the above, we will "solve" or at least improve/address points **2, 3, 4, 5**.

## Follow ups after this initial work

- Set up the owners/team based on `area`
- Based on the `area` automatically assign the issue to the respective owner/team
- More in depth definition of issue triage/from open to done, like as suggested in open-telemetry/opentelemetry-specification#3821

## Action items

- [x] Automatically generate `area` labels based on files inside `model/registry/*.yaml`
- [x] Automatically add the appropriate `area` label to new issues
- [x] Add ISSUE_TEMPLATES using the area labels as a drop-down
- [x] Add check to verify when new areas are introduced in the registry, so we can update the ISSUE_TEMPLATES area drop-down
- [x] Create new labels manually: `triage:needs-triage`, `area:other`, `area:new`

## Similar/related issues PRs

- #292
- #22
- open-telemetry/opentelemetry-specification#3821

### PR DESCRIPTION
**Pull request by joaopgrassi**
Fixes #773

## Changes

This PR adds several automations/improvements related to issue triage/project management and a bit of dev experience for users. 

The main things added are:

- Introduce concept of `area` - Those are the names of the `yaml` files inside `model/registry`
- Issue templates, that contain a drop-down for the `area`. 
- Scripts and actions to automate the process of:
  - Assigning the correct `area` label for new issues
  - Creating new labels when new areas are merged into the registry
  - Check when new areas are introduced in PRs, so the issue templates are always up-to-date (see example [here](https://github.com/joaopgrassi/test-actions-otel/pull/6))
  
As a follow up, we can: (out of scope of this PR)

- Improve the CONTRIBUTING.md to add the definition of the labels and talk about the issue templates a bit
- Check if we can use the new `area` concept and assign those to our GitHub teams, so issues are assigned to the proper responsible party, instead of always maintainers

This work is based on what the folks over on the collector contrib repo have. (Thanks @evan-bradley!)

You can check how this looks like now in this test repository https://github.com/joaopgrassi/test-actions-otel. Feel free to create issues to test, or create PRs to verify the checks.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        - area:user-agent
+        # End semconv area list
+  - type: textarea
+    attributes:
+      label: What happened?
+      description: Please provide as much detail as you reasonably can.
+      value: |
+        ## Description
+
+        ## Steps to Reproduce
```
Comment (lmolkova) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:56]: nit: do we need this item? we don't really have anything runnable or reproducible

Comment (joaopgrassi) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:56]: I was thinking on the case where someone uses an attribute, maybe update the semconv version, and then something goes wrong. That could be a "reproducible" situation. But fine also to remove it.

**Comment by pyohannes** [comment]
Given the fact that we now tag all new issues with `triage:needs-triage`, do you think it still makes sense to auto-assign issues to maintainers? It seems the auto-assignment to maintainers burdens them and didn't have the desired effect in the past.

For doing project management for messaging, I prefer to assign issues to folks who are actually working on it, and keep issues unassigned when nobody is working on it.

**Comment by joaopgrassi** [comment]
> Given the fact that we now tag all new issues with `triage:needs-triage`, do you think it still makes sense to auto-assign issues to maintainers? It seems the auto-assignment to maintainers burdens them and didn't have the desired effect in the past.

@pyohannes  Yes, that is true. In the linked issue I mentioned handling this as a follow up:

> Follow ups after this initial work
> 
>     Set up the owners/team based on area
>     Based on the area automatically assign the issue to the respective owner/team
>     More in depth definition of issue triage/from open to done, like as suggested in 
> 
> https://github.com/open-telemetry/opentelemetry-specification/issues/3821

We can debate whether to leave it unassigned or assign to the proper owners/team, like they do in the collector repos.

**Review by jsuereth — APPROVED** [review]

**Comment by pyohannes** [comment]
> We can debate whether to leave it unassigned or assign to the proper owners/team, like they do in the collector repos.

I'd be in favor of having it unassigned, the `area:*` and `needs-triage` labels should be enough to get the right people to look at it, but assigning it to proper owners for the given area should be fine too if that's preferred by others.

File: .github/ISSUE_TEMPLATE/bug_report.yaml (hunk 1 above)
Comment (lmolkova) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:56]: we can always remove it later depending on how useful it would be, definitely not blocking.

**Review by lmolkova — APPROVED** [review]

**Pull request merged**